master/internal/rm/kubernetesrm/spec_test.go

```diff
@@ -219,3 +220,123 @@ func TestAllPrintableCharactersInEnv(t *testing.T) {
 	require.Contains(t, actual, k8sV1.EnvVar{Name: "test2", Value: ""})
 	require.Contains(t, actual, k8sV1.EnvVar{Name: "func", Value: "f(x)=x"})
 }
+
+func TestDeterminedLabels(t *testing.T) {
+	// fill out task spec
+	taskSpec := tasks.TaskSpec{
+		Owner:       createUser(),
+		Workspace:   "test-workspace",
+		TaskType:    model.TaskTypeCommand,
+		TaskID:      model.NewTaskID().String(),
+		ContainerID: "container-id",
+	}
+
+	p := pod{
+		req: &sproto.AllocateRequest{
+			ResourcePool: "test-rp",
+		},
+		submissionInfo: &podSubmissionInfo{
+			taskSpec: taskSpec,
+		},
+	}
+
+	// define expectations
+	expectedLabels := map[string]string{
+		userLabel:         taskSpec.Owner.Username,
+		workspaceLabel:    taskSpec.Workspace,
+		resourcePoolLabel: p.req.ResourcePool,
+		taskTypeLabel:     string(taskSpec.TaskType),
+		taskIDLabel:       taskSpec.TaskID,
+		containerIDLabel:  taskSpec.ContainerID,
+	}
+
+	spec := p.configurePodSpec(make([]k8sV1.Volume, 1), k8sV1.Container{},
+		k8sV1.Container{}, make([]k8sV1.Container, 1), &k8sV1.Pod{}, "scheduler")
+
+	// confirm pod spec has required labels
+	require.NotNil(t, spec)
+	for expectedKey, expectedValue := range expectedLabels {
+		require.Equal(t, spec.ObjectMeta.Labels[expectedKey], expectedValue)
+	}
+}
+
+func TestTrialLabels(t *testing.T) {
+	// fill out task spec
+	experimentID := "5"
+	trialRequestID := model.NewTaskID().String()
+	taskSpec := tasks.TaskSpec{
+		Owner:       createUser(),
+		Workspace:   "test-workspace",
+		TaskType:    model.TaskTypeTrial,
+		TaskID:      experimentID + "." + trialRequestID,
```

Comment:
I wanted to form the TaskID using `internal.trialTaskID()` but ran into an import error. This way, if the function is changed meaningfully, the test will also fail and hopefully spec.go will also be updated.

I'm happy to make the change if it's worth moving `trialTaskID()` to another module; is there one you'd suggest?